test/with_dummyserver/test_connectionpool.py

```diff
@@ -987,5 +987,32 @@ def tell(self):
             self.assertTrue('Unable to record file position for' in str(e))
 
 
+class TestRetryPoolSize(HTTPDummyServerTestCase):
+    def setUp(self):
+        retries = Retry(
+            total=3,
+            backoff_factor=0.01,
+            raise_on_status=False,
+            status_forcelist=[404],
+        )
+        self.pool = HTTPConnectionPool(self.host, self.port, maxsize=10,
+                                       retries=retries, block=True)
+        self.addCleanup(self.pool.close)
+
+    def test_pool_size_retry(self):
+        self.pool.urlopen('GET', '/not_found', preload_content=False)
+        self.assertEquals(self.pool.num_connections, 1)
+
+
+class TestRedirectPoolSize(HTTPDummyServerTestCase):
+    def setUp(self):
+        self.pool = HTTPConnectionPool(self.host, self.port, maxsize=10, retries=5, block=True)
```

Comment:
It might be worth adding some more verbose retry setup here to guarantee that redirects are in fact turned on, just in case this API changes for any reason.